experiments/experiments_test.go

```diff
@@ -0,0 +1,151 @@
+package experiments
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	parser "github.com/Netflix/p2plab/cue/parser"
+	"github.com/Netflix/p2plab/metadata"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+)
+
+func TestExperimentDefinition(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	db, cleanup := newTestDB(t, "exptestdir")
+	defer func() {
+		if err := cleanup(); err != nil {
+			t.Fatal(err)
+		}
+	}()
+	var ids []string
+	t.Run("Experiment Creation And Retrieval", func(t *testing.T) {
+		sourceFiles := []string{
+			"../cue/cue.mod/p2plab_example1.cue",
+			"../cue/cue.mod/p2plab_example2.cue",
+		}
+		for _, sourceFile := range sourceFiles {
+			name := strings.Split(sourceFile, "/")
+			exp1 := newTestExperiment(t, sourceFile, name[len(name)-1])
+			ids = append(ids, exp1.ID)
+			exp2, err := db.CreateExperiment(ctx, exp1)
+			if err != nil {
+				t.Fatal(err)
+			}
+			require.Equal(t, exp1.ID, exp2.ID)
+			require.Equal(t, exp1.Status, exp2.Status)
+			require.Equal(t, exp1.Definition, exp2.Definition)
+			exp3, err := db.GetExperiment(ctx, exp1.ID)
+			if err != nil {
+				t.Fatal(err)
+			}
+			require.Equal(t, exp1.ID, exp3.ID)
+			require.Equal(t, exp1.Status, exp3.Status)
+			require.Equal(t, exp1.Definition, exp3.Definition)
+		}
+	})
+	t.Run("List Experiments", func(t *testing.T) {
+		experiments, err := db.ListExperiments(ctx)
+		if err != nil {
+			t.Fatal(err)
+		}
+		for _, experiment := range experiments {
+			if experiment.ID != ids[0] && experiment.ID != ids[1] {
+				t.Fatal("bad experiment id found")
+			}
+		}
+	})
+	t.Run("Update Experiments", func(t *testing.T) {
+		for _, id := range ids {
+			exp, err := db.GetExperiment(ctx, id)
+			if err != nil {
+				t.Fatal(err)
+			}
+			prevUpdateAt := exp.UpdatedAt
+			exp.Labels = append(exp.Labels, "test label")
+			exp, err = db.UpdateExperiment(ctx, exp)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if exp.UpdatedAt.Before(prevUpdateAt) {
```

Comment:
`require.True(t, exp.UpdatedAt.Before(prevUpdatedAt))`

Reply:
I changed this to instead of check `exp.UpdateAt.Before` to examine `exp.UpdateAt.After`